google-cloud-pubsub/samples/pubsub_publish_avro_records.rb

```diff
@@ -21,22 +21,28 @@ def publish_avro_records topic_id:, avsc_file:
 
   pubsub = Google::Cloud::Pubsub.new
 
-  topic = pubsub.topic topic_id
+  topic_admin = pubsub.topic_admin
+
+  topic = topic_admin.get_topic topic: pubsub.topic_path(topic_id)
+
+  encoding = topic.schema_settings.encoding
+
+  publisher = pubsub.publisher topic_id
 
   record = { "name" => "Alaska", "post_abbr" => "AK" }
 
-  if topic.message_encoding_binary?
+  if encoding == :BINARY
```

Comment:
Instead of the `if`...`elsif`...`else`, it's more Ruby-like to use `case`:

```ruby
case encoding
when :BINARY
  # Do something for binary
when :JSON
  # Do something for json
else
  # Do stuff for other encodings
end
```